app/src/main/java/ch/epfl/sweng/swenggolf/statistics/UserStats.java

```diff
@@ -0,0 +1,170 @@
+package ch.epfl.sweng.swenggolf.statistics;
+
+import android.util.Log;
+
+import com.google.firebase.database.Exclude;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+import ch.epfl.sweng.swenggolf.database.Database;
+import ch.epfl.sweng.swenggolf.database.DbError;
+import ch.epfl.sweng.swenggolf.database.ValueListener;
+import ch.epfl.sweng.swenggolf.profile.User;
+
+/**
+ * A class that allows to manage the statistics of a User.
+ */
+public class UserStats {
+    public static final Integer INITIAL_VALUE = 0;
+    private static final String LOG_KEY_STATISTICS = "USER_STATS";
+    private HashMap<String, Integer> stats;
+
+    /**
+     * Builds the stats with the default value for all of them.
+     */
+    public UserStats(){
+        setMap(new HashMap<String, Integer>());
+        for(Stats stat: Stats.values()){
+            put(stat, INITIAL_VALUE);
+        }
+    }
+
+    /**
+     * Adds a value for a specific stats in the Map.
+     * @param stat The stat that we want to add/change
+     * @param value The Integer value
+     */
+    public void put(Stats stat, Integer value){
+        stats.put(stat.toString(), value);
+    }
+
+    /**
+     * A getter of the Map that is needed by Firebase.
+     * @return The Map
+     */
+    public HashMap<String, Integer> getMap(){
+        return stats;
+    }
+
+    /**
+     * Returns the i-th key of the Map.
+     * @param i the index of the wanted element
+     * @return the corresponding key
+     */
+    @Exclude
+    public Stats getKey(Integer i){
+        return Stats.valueOf((String)stats.keySet().toArray()[i]);
+    }
+
+    /**
+     * A setter that is needed for the Firebase read function.
+     * @param map the HashMap with the stats
+     */
+    public void setMap(HashMap<String, Integer> map){
```

Comment:
I know it is a very little detail, but I think that it would be more readable to have the methods in a "logical" order (i.e. getters and setters of the same thing together, etc). But it's nice that you added some javadoc!

Reply:
Even the little details matter, it is changed :)